text/0000-convert-id.md

```diff
@@ -0,0 +1,203 @@
+both libcore and libstd.
+
+# Motivation
+[motivation]: #motivation
+
+## The identity function is useful
+
+While it might seem strange to have a function that just returns back the input,
+there are some cases where the function is useful.
+
+### Using `identity` to do nothing among a collection of mappers
+
+When you have collections such as maps or arrays of mapping functions like
+below and you watch to dispatch to those you sometimes need the identity
+function as a way of not transforming the input. You can use the identity
+function to achieve this.
+
+```rust
+// Let's assume that this and other functions do something non-trivial.
+fn do_interesting_stuff(x: u32) -> u32 { .. }
+
+// A dispatch-map of mapping functions:
+let mut map = HashMap::new();
+map.insert("foo", do_interesting_stuff);
+map.insert("bar", other_stuff);
+map.insert("baz", identity);
+```
+
+### Using `identity` as a no-op function in a conditional
+
+This reasoning also applies to simpler yes/no dispatch as below:
+
+```rust
+let mapper = if condition { some_manipulation } else { identity };
+
+// do more interesting stuff inbetween..
+
+do_stuff(42);
+```
+
+### Using `identity` to concatenate an iterator of iterators
+
+We can use the identity function to concatenate an iterator of iterators
+into a single iterator.
+
+```rust
+let vec_vec = vec![vec![1, 3, 4], vec![5, 6]];
+let iter_iter = vec_vec.into_iter().map(Vec::into_iter);
+let concatenated = iter_iter.flat_map(identity).collect::<Vec<_>>();
+assert_eq!(vec![1, 3, 4, 5, 6], concatenated);
+```
+
+### Using `identity` to keep the `Some` variants of an iterator of `Option<T>`
+
+We can keep all the maybe variants by simply `iter.filter_map(identity)`.
+
+```rust
+let iter = vec![Some(1), None, Some(3)].into_iter();
+let filtered = iter.filter_map(identity).collect::<Vec<_>>();
+assert_eq!(vec![1, 3], filtered);
+```
+
+### To be clear that you intended to use an identity conversion
+
+If you instead use a closure as in `|x| x` when you need an
+identity conversion, it is less clear that this was intentional.
+With `identity`, this intent becomes clearer.
+
+## The `drop` function as a precedent
+
+The `drop` function in `core::mem` is defined as `pub fn drop<T>(_x: T) { }`.
+The same effect can be achieved by writing `{ _x; }`. This presents us
+with a precendent that such trivial functions are considered useful and
+includable inside the standard library even though they can be written easily
+inside a user's crate.
+
+## Avoiding repetition in user crates
+
+Here are a few examples of the identity function being defined and used:
+
++ https://docs.rs/functils/0.0.2/functils/fn.identity.html
++ https://docs.rs/tool/0.2.0/tool/fn.id.html
++ https://github.com/hephex/api/blob/ef67b209cd88d0af40af10b4a9f3e0e61a5924da/src/lib.rs
+
+There's a smattering of more examples. To reduce duplication, it
+should be provided in the standard library as a common place it is defined.
```

Comment:
I don't think this argument is persuasive for functions.  There's a case for things like traits and structs having a common definition, since they're nominal and thus the commonality can improve interoperability (not that that place needs to be `std` even there).  But that's not the case for functions, where it doesn't matter if you used the `functils` or `tools` one.